src/Akka.Persistence.MongoDb/Journal/MongoDbJournal.cs

```diff
@@ -81,48 +94,68 @@ protected override void PreStart()
-                            .Ascending(entry => entry.Ordering));
+                        var modelWithOrdering = new CreateIndexModel<JournalEntry>(
+                            Builders<JournalEntry>
+                                .IndexKeys
+                                .Ascending(entry => entry.Ordering));
 
-                    collection.Indexes
-                        .CreateOne(modelWithOrdering);
+                        collection.Indexes
+                            .CreateOneAsync(modelWithOrdering, cancellationToken: unitedCts.Token)
+                            .Wait();
+                    }
                 }
 
                 return collection;
             });
 
             _metadataCollection = new Lazy<IMongoCollection<MetadataEntry>>(() =>
             {
-                var collection = _mongoDatabase.Value.GetCollection<MetadataEntry>(_settings.MetadataCollection);
-
-                if (_settings.AutoInitialize)
+                var (perCallCts, unitedCts) = CreatePerCalLCts();
+                using (perCallCts)
+                using (unitedCts)
                 {
-                    var modelWithAscendingPersistenceId = new CreateIndexModel<MetadataEntry>(
-                        Builders<MetadataEntry>
-                            .IndexKeys
-                            .Ascending(entry => entry.PersistenceId));
+                    var collection = _mongoDatabase.Value.GetCollection<MetadataEntry>(_settings.MetadataCollection);
+
+                    if (_settings.AutoInitialize)
+                    {
+                        var modelWithAscendingPersistenceId = new CreateIndexModel<MetadataEntry>(
+                            Builders<MetadataEntry>
+                                .IndexKeys
+                                .Ascending(entry => entry.PersistenceId));
 
-                    collection.Indexes
-                        .CreateOneAsync(modelWithAscendingPersistenceId, cancellationToken: CancellationToken.None)
+                        collection.Indexes
+                            .CreateOneAsync(modelWithAscendingPersistenceId, cancellationToken: unitedCts.Token)
                             .Wait();
-                }
+                    }
 
-                return collection;
+                    return collection;
+                }
             });
         }
 
-        public override async Task ReplayMessagesAsync(IActorContext context, string persistenceId, long fromSequenceNr, long toSequenceNr, long max, Action<IPersistentRepresentation> recoveryCallback)
+        private (CancellationTokenSource perCallCts, CancellationTokenSource untiedCts) CreatePerCalLCts()
+        {
+            var perCallCts = new CancellationTokenSource(_settings.CallTimeout);
+            var unitedCts =
+                CancellationTokenSource.CreateLinkedTokenSource(perCallCts.Token, _pendingCommandsCancellation.Token);
+            return (perCallCts, unitedCts);
```

Comment:
I think you can remove one token source by doing this:
```
var unitedCts = CancellationTokenSource.CreateLinkedTokenSource(_pendingCommandsCancellation.Token);
unitedCts.CancelAfter(_settings.CallTimeout);
return unitedCts;
```

Reply:
That'd help simplify things - I'll give that at try once you're finished with the review.

Reply:
Done